dioptas/model/loader/hdf5Loader.py

```diff
@@ -32,15 +32,15 @@ def __init__(self, filename):
         self.f = h5py.File(filename, 'r')
         self.image_sources = find_image_sources(self.f)
 
-        self.dataset = self.f[self.image_sources[0]]
-        self.series_max = self.dataset.shape[0]
+        self.__current_source = self.image_sources[0]
+        self.series_max = self.f[self.__current_source].shape[0]
 
     def get_image(self, ind):
-        return self.dataset[ind]
+        return self.f[self.__current_source][ind]
 
     def select_source(self, source):
-        self.dataset = self.f[source]
-        self.series_max = self.dataset.shape[0]
+        self.__current_source = source
+        self.series_max = self.f[source].shape[0]
```

Comment:
Do not store a reference to the dataset, since this causes tests to segfault.
Instead, store the source path and get the dataset late.

Reply:
why is this segfaulting?

Reply:
Honestly I don't know.

segfault seems to happen when Python is closing and maybe there's some references kept that delays closing the hdf5 file...

Reply:
I'll have a second look at this one

Reply:
Looking at this again, the segmentation fault looks to be in a function in libhdf5 related to virtual dataset.
But it's related to the use of a daemon thread in NewFileWatcher:

https://github.com/Dioptas/Dioptas/blob/develop/dioptas/model/util/NewFileWatcher.py#L79

And having reference kept when closing Python.
According to [threading doc](https://docs.python.org/3/library/threading.html#thread-objects):

> Daemon threads are abruptly stopped at shutdown. Their resources (such as open files, database transactions, etc.) may not be released properly.

The patch here avoids to keep a reference on the virtual dataset. It's main advantage is that it is local and simple.

There are other ways to workaround/fix this:
- Do not use a daemon thread, but then it needs to be explicitly stopped
- Avoid using a thread: I'm not sure why a thread is needed? Qt signals can be set to be asynchronous if it needs to be asynchronously handled.
- Make the Signal class in [model.util.signal](https://github.com/Dioptas/Dioptas/blob/ab77cf152a76fc52f15cee4f7f20183730f3b061/dioptas/model/util/signal.py#L24) keep weak reference on methods/functions: That would make it neat IMO by removing quite a few of reference that prevent object to be deleted, but it's a big change.